File: Nivel_2/Retos/Factorial/FactorialRecursivo.py
def factorial(n):
    if n < 0:
        return -1
    if n == 0 or n == 1:
        return 1
    return n * factorial(n - 1)

def main():
    n = 5  # Cambia este valor para calcular otros números
    resultado = factorial(n)
    if resultado == -1:
        print("El número debe ser no negativo.")
    else:
        print(f"El factorial de {n} es {resultado}")

if __name__ == "__main__":
    main()

File: Nivel_2/Retos/Factorial/FactorialRecursivo.java
public class FactorialRecursivo{
    public static void main(String[] args){
        int n = 5;
        System.out.println(String.format("El factorial de %d es %d", n, Factorial(n)));
    }

    public static long Factorial(int n){
        if(n < 0){return -1;}
        if(n == 0 || n == 1){return 1;}
        return n * Factorial(n - 1);
    }
}
